packages/polymorphic/src/Polymorphic/Polymorphic.ts

```diff
@@ -29,6 +29,7 @@ export const Polymorphic = <
   // (i.e. we don't know if `as="button"`, and if `type` is a valid prop)
 
   // If no `ref` arg was passed in, we use the plain render function
+  // @ts-expect-error FIXME: https://jira.mongodb.org/browse/LG-3410
```

Comment:
Change to `ts-ignore`. This prevents TS errors when running with `react^17.0.2`/`@type/react^16.9.56`(current `main` version)